packages/adblocker-electron-preload/src/index.ts

```diff
@@ -68,11 +53,11 @@ if (window === window.top && window.location.href.startsWith('devtools://') ===
     window.addEventListener(
       'DOMContentLoaded',
       () => {
-        DOM_MONITOR = new DOMMonitor((update) => {
+        DOM_MONITOR = new DOMMonitor(async (update) => {
```

Comment:
Regarding the error handling: since it is now async, the callback will return a promise. I wonder if this is a problem if the callback failed? From what I see, the caller (DOMMonitor) of the callback does not await the result. It would not see the error, nor wait for the processing here to be completed.

Maybe I'm missing something, but I assume the DOMMonitor is that class here: https://github.com/ghostery/adblocker/blob/40a4c8ba9e738b312284cddc6705b5b2a85864d8/packages/adblocker-content/src/index.ts#L176

Reply:
exactly - DOMMonitor relay on the fact the the it callback will execute sync. It is not only about error handling, but for example for `handleNewFeatures` which from what I see assumes the DOM was affected.

Reply:
I'll think of a way to address this, thank you for catching it!